### DISCUSSION TITLE
Integration with Github Actions

### DISCUSSION THREAD

**Discussion by Oleksii-QA** [webdriverio/webdriverio#8860] [Q&A]
Use WebdriverIO v7 with mocha framework and JavaScript. I integrate my tests into the Github action
Read the setup, studied the example in v7 branch https://github.com/webdriverio/jasmine-boilerplate/tree/v7

My github config works with:
.yml file
folder with test
package.json
wdio.conf.js

Is package-lock.json needed and another files?

**Comment by christian-bromann** [comment]
> Is package-lock.json needed and another files?

This is up to you to include that file. Read more on this here: https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json/

**Comment by Oleksii-QA** [comment]
@christian-bromann 
There are other questions. Please explain.
1. Does chrome only work in headless mode on ubuntu?
Use ubuntu-latest in .yml file and configuration in wdio.conf.js file:
maxInstances: 1,
      acceptInsecureCerts: true,
      browserName: "chrome",
      "goog:chromeOptions": {
        args: [
          "--headless",
          "--no-sandbox",
          "--disable-gpu",
          "--disable-dev-shm-usage",
          "--window-size=1920,1080",
2. Do I need to specify port 4444? I didn't specify the port. Test runs on different ports. As far as I understand, on the github servers.
3. Your example uses jasmine. Can mocha be used?
4. Use configuration in package.json:
"devDependencies": {
    "@wdio/cli": "^7.24.0",
    "@wdio/jasmine-framework": "^7.24.0",
    "@wdio/local-runner": "^7.24.0",
    "@wdio/mocha-framework": "^7.24.0",
    "@wdio/spec-reporter": "^7.24.0",
    "chromedriver": "^105.0.0",
    "wdio-chromedriver-service": "^7.3.2"
  }
Is this enough for normal test working?

5. Use spec reporter. Is it possible to download the test log?

**Reply by christian-bromann** [reply]
> Does chrome only work in headless mode on ubuntu?

Yes

> Do I need to specify port 4444? I didn't specify the port. Test runs on different ports. As far as I understand, on the github servers.

Depends how you set up WebdriverIO. If you use its testrunner with e.g. [Chromedriver service](https://www.npmjs.com/package/wdio-chromedriver-service) then you don't need to specify any port.

> Your example uses jasmine. Can mocha be used?

Totally!

> Is this enough for normal test working?

Yes.

> Use spec reporter. Is it possible to download the test log?

Yes by setting the [`outputDir`](https://webdriver.io/docs/options#outputdir) option either generally or as reporter option.
